In [1]:
# IMPORT libraries
import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings('ignore')

import matplotlib.pyplot as plt
%matplotlib inline

In [16]:
user = pd.read_csv('../data/processed/part-00000-46d48d05-1d5c-4096-a243-308a7216e2ff-c000.csv')
reviews = pd.read_csv("../data/processed/reviews.csv",index_col=[0])

In [3]:
user.columns

Index(['average_stars', 'compliment_cool', 'compliment_cute',
       'compliment_funny', 'compliment_hot', 'compliment_list',
       'compliment_more', 'compliment_note', 'compliment_photos',
       'compliment_plain', 'compliment_profile', 'compliment_writer', 'cool',
       'elite', 'fans', 'friends', 'funny', 'name', 'review_count', 'useful',
       'user_id', 'yelping_since'],
      dtype='object')

In [4]:
print("Businesses in the review dataset: ", reviews.business_id.nunique())

Businesses in the review dataset:  6219


In [23]:
business = pd.read_csv("../data/processed/business.csv")

In [6]:
len(business)

6217

There are total of 6217 businessees in New Orleans

In [7]:
len(reviews)

1095800

In [8]:
business.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6217 entries, 0 to 6216
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   address       5954 non-null   object 
 1   attributes    5722 non-null   object 
 2   business_id   6217 non-null   object 
 3   categories    6216 non-null   object 
 4   city          6217 non-null   object 
 5   hours         5059 non-null   object 
 6   is_open       6217 non-null   int64  
 7   latitude      6217 non-null   float64
 8   longitude     6217 non-null   float64
 9   name          6217 non-null   object 
 10  postal_code   6212 non-null   float64
 11  review_count  6217 non-null   int64  
 12  stars         6217 non-null   float64
 13  state         6217 non-null   object 
dtypes: float64(4), int64(2), object(8)
memory usage: 728.6+ KB


In [17]:
set(reviews.business_id.unique()) - set(business.business_id)

{'0', '6', nan}

In [18]:
reviews = reviews[reviews.business_id.str.len() != 22]

Removing unwanted values from reviews dataset

In [26]:
# doublecheck that there is only one review for each user_id, business_id combinations before pivoting to user x restaurant matrix
reviews.info()

<class 'pandas.core.frame.DataFrame'>
Index: 393218 entries,   to  I recommend not to go here and if your hotel offers the coupon tell them not to promote this restaurant they are cheaters and are a disgrace to cajun/creole cuisine.
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   business_id  2 non-null      object 
 1   cool         2 non-null      object 
 2   date         0 non-null      object 
 3   funny        0 non-null      float64
 4   review_id    0 non-null      object 
 5   stars        0 non-null      float64
 6   text         0 non-null      object 
 7   useful       0 non-null      float64
 8   user_id      0 non-null      object 
dtypes: float64(3), object(6)
memory usage: 38.1+ MB
